### PR TITLE
BUG: fix error on IBM POWER9 ppcle64 arch

### PR DESCRIPTION
**Pull request by Mon-ius**
Fix an error happens on **ppcle64** Arch Linux system for IBM Power9.

See details: https://github.com/InsightSoftwareConsortium/ITK/issues/1960.

## PR Checklist
- [ x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx [hunk 1]
```diff
@@ -30,7 +30,7 @@ bool
 WriteSmallBmp(const std::string & filename)
 {
   const char    bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
-                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, -1, 0 };
+                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, 255, 0 };
```
Comment (dzenanz) [inline Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:33]: maybe `static_cast<char>(-1)`?

Comment (blowekamp) [reply Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:33]: `char` may be signed or unsigned depending on the system. Perhaps changing the type to `unsigned char` in addition to this change  would be the most portable and explicit.

Comment (Mon-ius) [reply Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:33]: Just test `unsigned char` on IBM power9 ppcle64. For original `-1`:

```
/rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx: In function ‘bool {anonymous}::WriteSmallBmp(const string&)’:
/rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:40:20: error: invalid conversion from ‘const signed char*’ to ‘const char*’ [-fpermissive]
   bmp_output.write(bmp_raw, sizeof(bmp_raw));
                    ^~~~~~~
In file included from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/ostream:693,
                 from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/istream:39,
                 from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/sstream:38,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkMacro.h:53,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkLightObject.h:21,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkObject.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkRegion.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkImageRegion.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkImage.h:21,
                 from /rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:18:
/rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/bits/ostream.tcc:183:25: note:   initializing argument 1 of ‘std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::write(const _CharT*, std::streamsize) [with _CharT = char; _Traits = std::char_traits<char>; std::streamsize = long int]’
     write(const _CharT* __s, streamsize __n)
           ~~~~~~~~~~~~~~^~~
make[2]: *** [Modules/IO/BMP/test/CMakeFiles/ITKIOBMPTestDriver.dir/itkBMPImageIOTestExtension.cxx.o] Error 1
make[1]: *** [Modules/IO/BMP/test/CMakeFiles/ITKIOBMPTestDriver.dir/all] Error 2
```

Comment (Mon-ius) [reply Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:33]: Tried 3 ways to make it works.

```
  const  char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, static_cast<char>(-1) , 0 };
```

```
  const signed char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, -1 , 0 };

  bmp_output.write((char *)bmp_raw[0], sizeof(bmp_raw));
```

```
  const unsigned char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, 255 , 0 };

  bmp_output.write((char *)bmp_raw[0], sizeof(bmp_raw));
```

Comment (Mon-ius) [reply Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:33]: Trying to submit PR: https://github.com/InsightSoftwareConsortium/ITK/pull/1965

**Pull request closed without merging**